### ISSUE TITLE
"@container": ["@set", "@graph"] not producing a set of separate graphs

### ISSUE THREAD

**Issue by dlongley** [json-ld/json-ld.org#623]
```js
{
  "@context": {
    "@version": 1.1,
    "credential": {"@id": "cred:credential", "@container": ["@set", "@graph"]}
  },
  "credential": [{
    "@id": "foo:1",
    "foo:bar": "1"
  }, {
    "@id": "foo:2",
    "foo:bar": "2"
  }]
}
```

N-Quads output is:

```
<foo:1> <foo:bar> "1" _:b1 .
<foo:2> <foo:bar> "2" _:b1 .
_:b0 <cred:credential> _:b1 .
```

Expected N-Quads output is:

```
<foo:1> <foo:bar> "1" _:b1 .
<foo:2> <foo:bar> "2" _:b2 .
_:b0 <cred:credential> _:b1 .
_:b0 <cred:credential> _:b2 .
```

This happens whether or not `@set` is combined with the container, i.e. if just `@graph` is used for the `@container` and more than one value is included. Each value in a `@graph` container should be a separate graph. Note that if a disjoint graph is desired, the following syntax could be used (but is also temporarily broken):

```js
{
  "@context": {
    "@version": 1.1,
    "credential": {"@id": "cred:credential", "@container": ["@set", "@graph"]}
  },
  "credential": [
    [
      {"@id": "foo:1", "foo:bar": "1"},
      {"@id":"foo:3", "foo:bar": "3"}
    ],
    {
      "@id": "foo:2",
      "foo:bar": "2"
    }
  ]
}
```

N-Quads output is:

```
<foo:1> <foo:bar> "1" _:b1 .
<foo:2> <foo:bar> "2" _:b1 .
<foo:3> <foo:bar> "3" _:b1 .
_:b0 <cred:credential> _:b1 .
```

Expected N-Quads output is:

```
<foo:1> <foo:bar> "1" _:b1 .
<foo:2> <foo:bar> "2" _:b2 .
<foo:3> <foo:bar> "3" _:b1 .
_:b0 <cred:credential> _:b1 .
_:b0 <cred:credential> _:b2 .
```

**Comment by dlongley** [comment]
I filed an issue in jsonld.js as well -- though I suspect there's a bug in the algorithm and test suite.

**Comment by dlongley** [comment]
It looks like the test suite does not have sufficient coverage for this case. I think we've only got tests for single values.

**Comment by gkellogg** [comment]
> Note that if a disjoint graph is desired, the following syntax could be used (but is also temporarily broken

Not too keen on the array of arrays syntax; perhaps we simply don't support that use case? Need to explore the algorithm to see what makes the most sense.

**Comment by dlongley** [comment]
@gkellogg,

> Not too keen on the array of arrays syntax...

I think it just falls out automatically from the algorithm. I don't think we need to explicitly prohibit it. I do expect using it will be uncommon though.

**Comment by gkellogg** [comment]
Fixed in #624.
